File: import-maaz-clients.js
import { db } from './server/db.js';
import { clients } from './shared/schema.js';
import fs from 'fs';

async function importMaazClients() {
  console.log('🎯 IMPORTING MAAZ CLIENTS TO DATABASE');
  console.log('='.repeat(50));
  
  try {
    // Load the extracted Maaz client data
    const maazClientsData = JSON.parse(fs.readFileSync('maaz_clients_detailed.json', 'utf8'));
    console.log(`📊 Loaded ${maazClientsData.length} Maaz client records`);
    
    let importedCount = 0;
    let errorCount = 0;
    
    for (let i = 0; i < maazClientsData.length; i++) {
      const clientData = maazClientsData[i];
      
      try {
        // Extract key client information
        const clientId = clientData['CLIENT ID'] || '';
        const name = clientId ? `Client ${clientId}` : `Maaz Client ${i + 1}`;
        
        // Get phone from any phone-related field
        let phone = null;
        for (const [key, value] of Object.entries(clientData)) {
          if (key.toLowerCase().includes('phone') && value) {
            phone = String(value);
            break;
          }
        }
        
        // Get email from any email-related field  
        let email = null;
        for (const [key, value] of Object.entries(clientData)) {
          if (key.toLowerCase().includes('email') && value) {
            email = String(value);
            break;
          }
        }
        
        // Get status
        let status = (clientData['STATUS'] || 'prospect').toLowerCase();
        if (!['prospect', 'active', 'inactive', 'vip'].includes(status)) {
          status = 'prospect';
        }
        
        // Get segment/priority
        const segment = (clientData['CLIENT SEGMENT'] || 'medium').toLowerCase();
        let priority = 'medium';
        if (segment.includes('vip')) {
          priority = 'vip';
        } else if (segment.includes('high')) {
          priority = 'high';
        } else if (segment.includes('low')) {
          priority = 'low';
        }
        
        // Get interests (product references)
        let interests = clientData['REFERENCE'] || '';
        if (!interests) {
          // Look for any product/reference fields
          for (const [key, value] of Object.entries(clientData)) {
            if (['reference', 'product', 'model'].some(term => key.toLowerCase().includes(term)) && value) {
              interests = String(value);
              break;
            }
          }
        }
        
        // Get notes/comments
        let notes = clientData['COMMENTS'] || '';
        if (!notes) {
          // Combine other relevant fields as notes
          const noteParts = [];
          for (const [key, value] of Object.entries(clientData)) {
            if (!['CLIENT ID', 'STATUS', 'CLIENT SEGMENT', 'REFERENCE'].includes(key) && value) {
              noteParts.push(`${key}: ${value}`);
            }
          }
          notes = noteParts.slice(0, 3).join(' | '); // First 3 fields only
        }
        
        // Get boutique/location
        const location = clientData['BOUTIQUE'] || 'Phone sales Middle East';
        
        // Get dates
        const requestDateStr = clientData['REQUEST DATE'] || '';
        let lastInteraction = null;
        if (requestDateStr) {
          try {
            // Parse date (format: 29/8/2025)
            const [day, month, year] = requestDateStr.split('/');
            lastInteraction = new Date(`${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`);
          } catch (e) {
            // Date parsing failed, leave as null
          }
        }
        
        // Calculate lead score based on available data
        let leadScore = 50; // Base score
        if (status === 'active') leadScore += 20;
        if (priority === 'vip') leadScore += 30;
        else if (priority === 'high') leadScore += 15;
        if (interests) leadScore += 10;
        if (phone) leadScore += 10;
        if (email) leadScore += 10;
        
        // Insert client into database
        await db.insert(clients).values({
          name,
          phone,
          email,
          status,
          priority,
          interests,
          location,
          notes,
          leadScore,
          conversionProbability: leadScore > 70 ? 0.7 : 0.5,
          lastInteraction,
          tags: ['maaz', 'luxury_watches', 'vacheron_constantin'],
          totalInteractions: 1,
          followUpRequired: true,
          followUpDate: new Date(Date.now() + 24 * 60 * 60 * 1000) // Tomorrow
        }).onConflictDoNothing();
        
        importedCount++;
        
        if ((i + 1) % 50 === 0) {
          console.log(`   📈 Imported ${i + 1}/${maazClientsData.length} clients...`);
        }
        
      } catch (error) {
        errorCount++;
        console.log(`   ❌ Error importing client ${i + 1}: ${error.message.substring(0, 100)}`);
        continue;
      }
    }
    
    console.log(`\n✅ IMPORT COMPLETE!`);
    console.log(`   📊 Successfully imported: ${importedCount} clients`);
    console.log(`   ❌ Errors: ${errorCount} clients`);
    console.log(`   🎯 Total Maaz clients in database: ${importedCount}`);
    
    // Verify import
    const maazCount = await db
      .select()
      .from(clients)
      .where(sql`'maaz' = ANY(tags)`);
    
    console.log(`   ✅ Database verification: ${maazCount.length} Maaz-tagged clients found`);
    
  } catch (error) {
    console.error('❌ Import failed:', error.message);
  }
}

// Run the import
importMaazClients().catch(console.error);

File: import_maaz_clients.py
import json
import psycopg2
from datetime import datetime
import os

def import_maaz_clients_to_database():
    """Import all Maaz client data to the PostgreSQL database"""
    
    print("🎯 IMPORTING MAAZ CLIENTS TO DATABASE")
    print("="*50)
    
    # Load the extracted Maaz client data
    try:
        with open('maaz_clients_detailed.json', 'r', encoding='utf-8') as f:
            maaz_clients = json.load(f)
        print(f"📊 Loaded {len(maaz_clients)} Maaz client records")
    except FileNotFoundError:
        print("❌ maaz_clients_detailed.json not found. Please run extract_maaz_clients.py first.")
        return
    
    # Connect to database
    try:
        conn = psycopg2.connect(os.environ['DATABASE_URL'])
        cursor = conn.cursor()
        print("✅ Connected to database")
    except Exception as e:
        print(f"❌ Database connection failed: {str(e)}")
        return
    
    # Import each client
    imported_count = 0
    error_count = 0
    
    for i, client_data in enumerate(maaz_clients, 1):
        try:
            # Extract key client information
            client_id = client_data.get('CLIENT ID', '')
            name = f"Client {client_id}" if client_id else f"Maaz Client {i}"
            
            # Get phone from any phone-related field
            phone = None
            for key, value in client_data.items():
                if 'phone' in key.lower() and value:
                    phone = str(value)
                    break
            
            # Get email from any email-related field  
            email = None
            for key, value in client_data.items():
                if 'email' in key.lower() and value:
                    email = str(value)
                    break
            
            # Get status
            status = client_data.get('STATUS', 'prospect').lower()
            if status not in ['prospect', 'active', 'inactive', 'vip']:
                status = 'prospect'
            
            # Get segment/priority
            segment = client_data.get('CLIENT SEGMENT', 'medium').lower()
            priority = 'medium'
            if 'vip' in segment:
                priority = 'vip'
            elif 'high' in segment:
                priority = 'high'
            elif 'low' in segment:
                priority = 'low'
            
            # Get interests (product references)
            interests = client_data.get('REFERENCE', '')
            if not interests:
                # Look for any product/reference fields
                for key, value in client_data.items():
                    if any(term in key.lower() for term in ['reference', 'product', 'model']) and value:
                        interests = str(value)
                        break
            
            # Get notes/comments
            notes = client_data.get('COMMENTS', '')
            if not notes:
                # Combine other relevant fields as notes
                note_parts = []
                for key, value in client_data.items():
                    if key not in ['CLIENT ID', 'STATUS', 'CLIENT SEGMENT', 'REFERENCE'] and value:
                        note_parts.append(f"{key}: {value}")
                notes = " | ".join(note_parts[:3])  # First 3 fields only
            
            # Get boutique/location
            location = client_data.get('BOUTIQUE', 'Phone sales Middle East')
            
            # Get dates
            request_date_str = client_data.get('REQUEST DATE', '')
            last_interaction = None
            if request_date_str:
                try:
                    # Parse date (format: 29/8/2025)
                    day, month, year = request_date_str.split('/')
                    last_interaction = f"{year}-{month.zfill(2)}-{day.zfill(2)}"
                except:
                    pass
            
            # Calculate lead score based on available data
            lead_score = 50  # Base score
            if status == 'active': lead_score += 20
            if priority == 'vip': lead_score += 30
            elif priority == 'high': lead_score += 15
            if interests: lead_score += 10
            if phone: lead_score += 10
            if email: lead_score += 10
            
            # Insert client into database
            insert_query = """
                INSERT INTO clients (
                    name, phone, email, status, priority, interests, location, 
                    notes, lead_score, conversion_probability, last_interaction,
                    tags, total_interactions, follow_up_required
                ) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
                ON CONFLICT DO NOTHING
            """
            
            cursor.execute(insert_query, (
                name,
                phone,
                email, 
                status,
                priority,
                interests,
                location,
                notes,
                lead_score,
                0.7 if lead_score > 70 else 0.5,  # Conversion probability
                last_interaction,
                ['maaz', 'luxury_watches', 'vacheron_constantin'],  # Tags
                1,  # Total interactions
                True  # Follow up required
            ))
            
            imported_count += 1
            
            if i % 50 == 0:
                print(f"   📈 Imported {i}/{len(maaz_clients)} clients...")
                
        except Exception as e:
            error_count += 1
            print(f"   ❌ Error importing client {i}: {str(e)[:100]}")
            continue
    
    # Commit changes
    try:
        conn.commit()
        print(f"\n✅ IMPORT COMPLETE!")
        print(f"   📊 Successfully imported: {imported_count} clients")
        print(f"   ❌ Errors: {error_count} clients")
        print(f"   🎯 Total Maaz clients in database: {imported_count}")
        
        # Verify import
        cursor.execute("SELECT COUNT(*) FROM clients WHERE 'maaz' = ANY(tags)")
        maaz_count = cursor.fetchone()[0]
        print(f"   ✅ Database verification: {maaz_count} Maaz-tagged clients found")
        
    except Exception as e:
        print(f"❌ Commit failed: {str(e)}")
        conn.rollback()
    
    finally:
        cursor.close()
        conn.close()
        print("🔐 Database connection closed")

if __name__ == "__main__":
    import_maaz_clients_to_database()